src/combinatorics/simple_graph/order.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright (c) 2021 Eric. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import combinatorics.simple_graph.subgraph
+
+/-!
+# Order relation on simple graphs
+
+This file gives `simple_graph` a `bounded_lattice` instance under the `is_subgraph` relation, and
+proves a couple lemmata about how this relation interacts with `simple_graph.subgraph.spanning_coe`.
+
+# Main definitions:
+
+* `simple_graph.is_subgraph`: A `simple_graph` is a subgraph of another if its `adj` is dominated by
+  the other graph's.
+
+* `bounded_lattice` instance: With this definition, `simple_graph` forms a `bounded_lattice`.
+
+* `simple_graph.to_subgraph`: If a `simple_graph` is a subgraph of another, then you can turn it
+  into a member of the larger graph's `simple_graph.subgraph` type.
+
+-/
+
+namespace simple_graph
+
+variable {V : Type*}
+
+/-- The relation that one `simple_graph` is a subgraph of another. -/
+def is_subgraph (x y : simple_graph V) : Prop := ∀ ⦃v w : V⦄, x.adj v w → y.adj v w
+
+/-- The union of two `simple_graph`s. -/
+def union (x y : simple_graph V) : simple_graph V :=
+{ adj := x.adj ⊔ y.adj,
+  sym := λ v w h, by rwa [sup_apply, sup_apply, x.adj_comm, y.adj_comm] }
+
+/-- The intersection of two `simple_graph`s. -/
+def inter (x y : simple_graph V) : simple_graph V :=
+{ adj := x.adj ⊓ y.adj,
+  sym := λ v w h, by rwa [inf_apply, inf_apply, x.adj_comm, y.adj_comm] }
+
+instance : has_union (simple_graph V) := ⟨union⟩
+instance : has_inter (simple_graph V) := ⟨inter⟩
+
+instance : bounded_lattice (simple_graph V) :=
+{ le := is_subgraph,
+  sup := union,
+  inf := inter,
+  top := complete_graph V,
+  bot := empty_graph V,
```

Comment:
I think you should remove `complete_graph` and `empty_graph` entirely, and in the file they're currently defined in replace them with `has_top` and `has_bot` (with docstrings indicating they are the empty and full graphs).

Reply:
mentioned on Zulip to see if others have issues with it

Reply:
I think there's value in having `complete_graph` and `empty_graph`, but maybe just as aliases of `top` and `bot`

Reply:
What do you mean by "alias"?

Reply:
Any of these really - I agree the intended form should be using the lattice instances, but for pedagogical reasons I feel like it would be nice to have them; similar to `vector_space`.

Reply:
Abbreviations like `vector_space` work much better for types (where they're completely transparent) than for data.

So maybe the thing to do is just add simp lemmas that say `complete_graph = \top` and `empty_graph = \bot`, and mention in the docstring for the two defs to prefer the notation?

Reply:
Sounds good!

Reply:
Pedagogy is also why I was thinking `complete_graph` is nice to have.  I like the idea of having these definitions that help you find these objects, with documentation suggesting what to use instead.